numpy/ma/mrecords.py

```diff
@@ -659,13 +659,12 @@ def openfile(fname):
         return fname
     # Try to open the file and guess its type
     try:
-        f = open(fname)
+        with open(fname) as f:
+            if f.readline()[:2] != "\\x":
+                f.seek(0, 0)
+                return f
```

Comment:
This changes the meaning - you can't move the with inside the try except.

There's nothing to stop you using:
```
try:
   f = open(fname)
except IoError:
    ...

# now use the with statement
with f:
   ...
```

Reply:
The function's intent was to return an open file handle only, so not attempting to fix the LGTM alert on this